sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3718,6 +3720,94 @@ def _rsync_down(args) -> None:
+                                 serve_lib.ServiceComponent],
+                             replica_id: Optional[int]) -> None:
+        ssh_credentials = backend_utils.ssh_credential_from_yaml(
+            controller_handle.cluster_yaml, controller_handle.docker_user)
+        runner = command_runner.SSHCommandRunner(
+            controller_handle.head_ip,
+            port=controller_handle.head_ssh_port,
+            **ssh_credentials,
+        )
+        sky_logs_directory = os.path.expanduser(constants.SKY_LOGS_DIRECTORY)
+        run_timestamp = backend_utils.get_run_timestamp()
+        sync_down_all_components = service_component is None
+
+        if (sync_down_all_components or
+                service_component == serve_utils.ServiceComponent.REPLICA):
+            replica_id_arg = replica_id if replica_id is not None \
+                else serve_constants.NO_REPLICA_ID_SPECIFIED
+            prepare_code = (
+                serve_utils.ServeCodeGen.prepare_replica_logs_for_download(
+                    service_name, run_timestamp, replica_id_arg))
+            backend = backend_utils.get_backend_from_handle(controller_handle)
+            assert isinstance(backend, backends.CloudVmRayBackend)
+            assert isinstance(controller_handle,
+                              backends.CloudVmRayResourceHandle)
+            logger.info('Preparing replica logs for download on controller')
+            prepare_returncode = self.run_on_head(controller_handle,
+                                                  prepare_code,
+                                                  require_outputs=False,
+                                                  stream_logs=False)
+            subprocess_utils.handle_returncode(
+                prepare_returncode, prepare_code,
+                'Failed to prepare replica logs to sync down.')
+            remote_service_dir_name = (
+                serve_utils.generate_remote_service_dir_name(service_name))
+            dir_for_download = os.path.join(remote_service_dir_name,
+                                            run_timestamp)
+            logger.info('Downloading the replica logs...')
+            runner.rsync(source=dir_for_download,
+                         target=sky_logs_directory,
+                         up=False,
+                         stream_logs=False)
+            remove_code = (
+                serve_utils.ServeCodeGen.remove_replica_logs_for_download(
+                    service_name, run_timestamp))
+            remove_returncode = self.run_on_head(controller_handle,
+                                                 remove_code,
+                                                 require_outputs=False,
+                                                 stream_logs=False)
+            subprocess_utils.handle_returncode(
+                remove_returncode, remove_code,
+                'Failed to remove the replica logs for '
+                'download on the controller.')
```

Comment:
pls try-catch and convert it to RuntimeError; ref https://github.com/skypilot-org/skypilot/blob/e51c2fb4b7e5f06cde6070ad0c758df3bc888026/sky/core.py#L841-L848

Reply:
same for above